workbench-core/datasets/src/wbcDataSetsAuthorizationPlugin.ts

```diff
@@ -96,8 +114,30 @@ export class WbcDataSetsAuthorizationPlugin implements DataSetsAuthorizationPlug
     throw new Error('Method not implemented.');
   }
 
-  public async getAllDataSetAccessPermissions(datasetId: string): Promise<PermissionsResponse> {
-    throw new Error('Method not implemented.');
+  public async getAllDataSetAccessPermissions(
+    datasetId: string,
+    pageToken?: string
+  ): Promise<PermissionsResponse> {
+    const identityResponse = await this._authorizer.getIdentityPermissionsBySubject({
+      subjectId: datasetId,
+      subjectType: dataSetSubjectType,
+      paginationToken: pageToken
+    });
+    const identityPermissions = _.filter(
+      identityResponse.data.identityPermissions,
+      (v: IdentityPermission) => v.action === 'READ' || v.action === 'UPDATE'
+    );
+    const permissionsResponse = this._identityPermissionsToPermissionsResponse(identityPermissions);
+
+    const permissionsCount = permissionsResponse.length;
+    if (permissionsCount > 1) {
+      throw new InvalidPermissionError(
+        `Expected a single permissions response, but got ${permissionsCount}.`
+      );
+    }
+
+    permissionsResponse[0].pageToken = identityResponse.paginationToken;
```

Comment:
This will fail if permissions count is empty. Probably want `if (permissionsCount != 1)` instead to cover the 0 case.

Reply:
Good catch, but now I realize that I need integration tests for that case!